src/components/SurvivorSelection.tsx

```diff
@@ -0,0 +1,40 @@
+import "../styles/survivor.css";
+import { Survivor, UserSelection } from "../utilities/types";
+
+export interface Props {
+  selectedSurvivor: Survivor;
+  survivorList: Survivor[];
+  setSelectedSurvivor: React.Dispatch<React.SetStateAction<Survivor>>;
+  setUserSelection: React.Dispatch<React.SetStateAction<UserSelection>>;
+}
+
+const SurvivorSelection = ({
+  selectedSurvivor,
+  survivorList,
+  setSelectedSurvivor,
+  setUserSelection
+}: Props) => {
+
+  console.log(selectedSurvivor);
+  
+  const handleSelection = (survivor: Survivor) => {
+    setSelectedSurvivor(survivor)
+    setUserSelection((prevUserSelection) => ({...prevUserSelection, id:Date.now(), userSurvivor: survivor}))
+  }
```

Comment:
something worth thinking about whenever you are pulling in a parent's setState `React.Dispatch` as a prop (like above, with `setSelectedSurvivor` and `setUserSelection`)

there's no particular reason this function can't be defined in the parent, then passed down into the child; as opposed to passing down the setState to the child, then defining in the child

so that would look like

parent:
```
const handleSurvivorSelection = (survivor: Survivor) => {
    setSelectedSurvivor(survivor);
    setUserSelection((prevUserSelection) => ({...prevUserSelection, id:Date.now(), userSurvivor: survivor}));
  }

...

# which allows you to not have to pass setStates, where technically anything can happen afterwards
# but instead, intentionally send over a self-descriptive function
<SurvivorSelection
    survivorList={survivorList}
    selectedSurvivor={selectedSurvivor}
    handleSurvivorSelection={handleSurvivorSelection}
/>
```

Reply:
is the benefit just so that the child component isn't exposed to the setstates, therefore less risk of potential issues? id assume the parent would be flooded with these handle functions if I did this with other components. My thought process was that the child component and all the function functionalities would be encapsolated in there, and only updated states would be fed up to parent.

Reply:
> is the benefit just so that the child component isn't exposed to the setstates, therefore less risk of potential issues?

correct. at the end of the day, we want to practice separation of concerns: the only reason the child has these setStates in the first place is only to do survivor selection. so why would we want to give it the freedom to do anything else, other than `handleSurvivorSelection`?

> id assume the parent would be flooded with these handle functions if I did this with other components. My thought process was that the child component and all the function functionalities would be encapsolated in there, and only updated states would be fed up to parent.

while that is a fair assessment, the alternative is to flood the children components with `setStates` where their purpose is ambiguous

the philosophy behind this is that:
- the parent contains the data layer + data logic
- the child only knows strictly what it needs to know

the original code would have:
- the parent contains the data layer
- the child runs the data logic, feeding the results to the parent

at this level, it's largely a matter of styling / preference

but in practice, flooding the parent with handle functions is what you'd actually want -> this is because as you scale, you can probably move some of these handle functions into stores, hooks, or any similar pattern to lift the state up beyond local-state. this can still be done with the original implementation pattern, but is substantially harder since the data layer / logic are not all in 1 place

Reply:
i do want to note tho, that i think your current concern with flooding the parent with handle functions is due to the way you've got the survivor data layers & logic in App.tsx, and how itll scale as you keep implementing

context is that application flow is:
- survivor selection page
- shop sim page
- results page

realistically it should be something like:
- App: conditionally renders the appropriate page, and only cares about navigating through the app flow
- SurvivorSelection: all the survivor related data layers & logic live here